synapse/storage/databases/main/event_push_actions.py

```diff
@@ -134,48 +153,100 @@ def _get_unread_counts_by_receipt_txn(
     def _get_unread_counts_by_pos_txn(self, txn, room_id, user_id, stream_ordering):
 
         # First get number of notifications.
-        # We don't need to put a notif=1 clause as all rows always have
-        # notif=1
-        sql = (
-            "SELECT count(*)"
-            " FROM event_push_actions ea"
-            " WHERE"
-            " user_id = ?"
-            " AND room_id = ?"
-            " AND stream_ordering > ?"
+        # We need to look specifically for events with notif = 1 because otherwise we'll
```

Comment:
I wonder if we can and should flatten these three lookups into one query

```
SELECT count(case when notif>0 then 1 end), count(case when highlight>0 then 1 end), // etc
```

may not be worthwhile though.

Reply:
I tried to do that initially and it quickly became a bit of a timesink, which is why I'm doing it separately here.

Reply:
I'd be interested if you could give me an idea of why it didn't work? it feels like it should just work, and would be a worthwhile optimisation.

Agreed that you shouldn't spend ages on it if it isn't trivial though.

Reply:
I guess you'd need separate queries on `event_push_actions` and `event_push_summary`

Reply:
Turns out I misunderstood what you meant and didn't know `CASE` was a thing in SQL.